source/jormungandr/tests/routing_tests_experimental.py

```diff
@@ -76,7 +76,7 @@ def test_first_and_last_section_multi_modes(self):
         # journey count = 18 / direct_path_call_count = 26 / routing_matrix_call_count = 20
         # get_directpath_count_by_mode(response, 'walking') == 5
         # get_directpath_count_by_mode(response, 'bike') == 5
-        assert len(response["journeys"]) == 10
+        assert len(response["journeys"]) == 13
```

Comment:
I don't get why there are more journeys here.

Reply:
aha i took me a lot of time to understand too.
 Notice that there is a `min_nb_jouneys=10` in the request and there is a condition that stops the search when there is no new qualified journeys(https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/scenarios/new_default.py#L984)

what happened before the fix is, with the bad similar journey filter condition, the loop is stopped at 5th iteration because it deleted some journeys that shouldn't be deleted. The new filter condition will delete less journeys and change the loop number(7 iterations) then more journeys are found.